cmd/argo-watcher/argo_status_updater_test.go

```diff
@@ -0,0 +1,399 @@
+			Id:  "test-id",
+			App: "test-app",
+		}
+
+		// mock calls
+		apiMock.EXPECT().GetApplication(task.App).Return(nil, fmt.Errorf("applications.argoproj.io \"test-app\" not found"))
+		stateMock.EXPECT().SetTaskStatus(task.Id, models.StatusAppNotFoundMessage, "ArgoCD API Error: applications.argoproj.io \"test-app\" not found")
+
+		// run the rollout
+		updater.WaitForRollout(task)
+	})
+
+	t.Run("Status Updater - ArgoCD unavailable", func(t *testing.T) {
+		// mocks
+		apiMock := mock.NewMockArgoApiInterface(ctrl)
+		metricsMock := mock.NewMockMetricsInterface(ctrl)
+		stateMock := mock.NewMockState(ctrl)
+
+		// argo manager
+		argo := &Argo{}
+		argo.Init(stateMock, apiMock, metricsMock)
+
+		// argo updater
+		updater := &ArgoStatusUpdater{}
+		updater.Init(*argo, 1, 0*time.Second, "test-registry")
+
+		// prepare test data
+		task := models.Task{
+			Id:  "test-id",
+			App: "test-app",
+		}
+
+		// mock calls
+		apiMock.EXPECT().GetApplication(task.App).Return(nil, fmt.Errorf(argoUnavailableErrorMessage))
+		stateMock.EXPECT().SetTaskStatus(task.Id, models.StatusAborted, "ArgoCD API Error: connect: connection refused")
+
+		// run the rollout
+		updater.WaitForRollout(task)
+	})
+
+	t.Run("Status Updater - Application API error", func(t *testing.T) {
+		// mocks
+		apiMock := mock.NewMockArgoApiInterface(ctrl)
+		metricsMock := mock.NewMockMetricsInterface(ctrl)
+		stateMock := mock.NewMockState(ctrl)
+
+		// argo manager
+		argo := &Argo{}
+		argo.Init(stateMock, apiMock, metricsMock)
+
+		// argo updater
+		updater := &ArgoStatusUpdater{}
+		updater.Init(*argo, 1, 0*time.Second, "test-registry")
+
+		// prepare test data
+		task := models.Task{
+			Id:  "test-id",
+			App: "test-app",
+		}
+
+		// mock calls
+		apiMock.EXPECT().GetApplication(task.App).Return(nil, fmt.Errorf("Unexpected failure"))
+		metricsMock.EXPECT().AddFailedDeployment(task.App)
+		stateMock.EXPECT().SetTaskStatus(task.Id, models.StatusFailedMessage, "ArgoCD API Error: Unexpected failure")
+
+		// run the rollout
+		updater.WaitForRollout(task)
+	})
+
+	t.Run("Status Updater - Application not available", func(t *testing.T) {
+		// mocks
+		apiMock := mock.NewMockArgoApiInterface(ctrl)
+		metricsMock := mock.NewMockMetricsInterface(ctrl)
+		stateMock := mock.NewMockState(ctrl)
+
+		// argo manager
+		argo := &Argo{}
+		argo.Init(stateMock, apiMock, metricsMock)
+
+		// argo updater
+		updater := &ArgoStatusUpdater{}
+		updater.Init(*argo, 1, 0*time.Second, "test-registry")
+
+		// prepare test data
+		task := models.Task{
+			Id:  "test-id",
+			App: "test-app",
+			Images: []models.Image{
+				{
+					Image: "ghcr.io/shini4i/argo-watcher",
+					Tag:   "dev",
+				},
+			},
+		}
+
+		// application
+		application := models.Application{}
+		application.Status.Summary.Images = []string{"test-image:v0.0.1"}
+
+		// mock calls
+		apiMock.EXPECT().GetApplication(task.App).Return(&application, nil).Times(2) // we call 1st time to check the status and 2nd time to
```

Comment:
It seems that this comment is not finalized. And the same thing for other lines of code that have the same comment.